javascript/ql/src/semmle/javascript/frameworks/NodeJSLib.qll

```diff
@@ -506,4 +506,295 @@ module NodeJSLib {
     }
   }
 
+  /**
+   * A data flow node that is a Node.js HTTP or HTTPS client request.
+   */
+  abstract class ClientRequest extends DataFlow::TrackedNode {
+    abstract DataFlow::Node getOptions();
+  }
+  
+  /**
+   * A data flow node that is a Node.js HTTP client request.
+   */
+  class HttpRequest extends ClientRequest {
+    HttpRequest() {
+      exists(DataFlow::MethodCallNode mcn |
+        mcn = this and
+        DataFlow::moduleImport("http").flowsTo(mcn.getReceiver()) and
+        mcn.getMethodName() = "request"
+      )
+    }
+    
+    override DataFlow::Node getOptions() {
+      result = this.(DataFlow::MethodCallNode).getArgument(0)
+    }
+  }
+  
+  /**
+   * A data flow node that is a Node.js HTTPS client request.
+   */
+  class HttpsRequest extends ClientRequest {
+    HttpsRequest() {
+      exists(DataFlow::MethodCallNode mcn |
+       mcn = this and
+        DataFlow::moduleImport("https").flowsTo(mcn.getReceiver()) and
+        mcn.getMethodName() = "request"
+      )
+    }
+    
+    override DataFlow::Node getOptions() {
+      result = this.(DataFlow::MethodCallNode).getArgument(0)
+    }
+  }
+  
+  /**
+   * A data flow node that is  the parameter of a result callback for an HTTP or HTTPS request made through Node.js.
+   */
+  abstract class RequestCallbackParam extends DataFlow::TrackedNode, RemoteFlowSource {
```

Comment:
This overlaps with `RequestSource`; same questions as for `ClientRequest` above.